### ISSUE TITLE
The LiteBuffer Created From This Has Browser Compatibility Issues

### ISSUE THREAD

**Issue by kbahr** [rsocket/rsocket-js#19]
https://github.com/rsocket/rsocket-js/blob/1d29ddb1efb868b84f0756cf5ed9ff717c901289/packages/rsocket-core/src/LiteBuffer.js#L121

I'm getting these objects from the WebSocket client and attempting to deserialize the `data` field of the incoming Payload through Google's Protobuf deserializer. It's throwing up an error that the input is not convertible to a Uint8Array (
https://github.com/google/protobuf/issues/1319). I've commented there also but to spare the reader clicking through, here is where the magic happens:
```
jspb.utils.byteSourceToUint8Array = function(a) {
    if (a.constructor === Uint8Array) return a;
    if (
        a.constructor === ArrayBuffer ||
        a.constructor === Buffer ||
        a.constructor === Array
    )
        return new Uint8Array(a);
    if (a.constructor === String)
        return goog.crypt.base64.decodeStringToUint8Array(a);
    goog.asserts.fail("Type not convertible to Uint8Array.");
    return new Uint8Array(0);
};
```

It appears that in Node environments, the `a.constructor === Buffer` check passes but in a browser (Chrome and Safari at least), it does not, falling through the Assert.

**Comment by lexs** [comment]
Thanks for letting me know, I have a diff-in-process that refactors all of code to have proper flow typing and I'll make sure to add a test like this.

**Comment by kbahr** [comment]
I like it - I'll pull this down and try it out tomorrow to verify.

**Comment by lexs** [comment]
@kbahr Did you have time to test this?

**Comment by kbahr** [comment]
@lexs I made a couple of comments on the PR or so I thought - I ran into an
issue where in BufferUtils.createBuffer gave me a LiteImpl which fails the
check `value instanceof Buffer` within BufferEncoder.byteLength.

**Comment by kbahr** [comment]
I see what I did - I failed to hit "Submit Review".

**Comment by viglucci** [comment]
Closing as I believe this was resolved by either #20 or #102
